src/relay/pass/anf.cc

```diff
@@ -0,0 +1,143 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file src/tvm/relay/pass/anf.cc
+ * \brief Transform the program into A Normal Form.
+ */
+#include <tvm/relay/expr_functor.h>
+#include "./type_visitor.h"
+#include "tvm/relay/pass.h"
+#include "./let_list.h"
+namespace tvm {
+namespace relay {
+
+class ANFMutator : private ExprMutator {
+  LetList ll;
+  template<typename K, typename V>
+  using Map = std::shared_ptr<std::unordered_map<K, V, NodeHash, NodeEqual> >;
```

Comment:
shared_ptr not necessary here

Reply:
it sort of is. when I recurse, I need to pass in all the map, as the recursed is under the same scope.

Reply:
(Granted, passing raw map work just as fine, but map is not small, so I will like to share them

Reply:
likely you can get away by passing pointer